delira/data_loading/sampler/batch.py

```diff
@@ -0,0 +1,65 @@
+from delira.data_loading.sampler.abstract import AbstractSampler
+
+
+class BatchSampler(object):
```

Comment:
I'm not sure about the purpose of this class. It does something similar as the `DataLoader` without resolving the internal dicts. What is the usecase of this sampler? (Could we replace the `DataLoader` with this sampelr?)

Reply:
No we can't. This sampler builds batches of indices (The sampler classes only return single indices now). The DataLoader mainly takes these indices, loads the corresponding data and combines them to single arrays.

The pipeline is as follows:
A sampler is created and wrapped by this class (to get batches of indices). It lives in the main process and sends these batches to the worker processes. These processes now load the data, combine them in batches and send them back to the main process.

We could instead include the batch-building in the samplers, but I really prefer it this way, since including the logic of dropping the last batch (and especially the checks for that) complicate the sampling algorithms beyond necessity.